### PR TITLE
fix: keep custom http client with converter

### LINKED ISSUE #271: httpClient is not used with .withConverter()
Describe the bug
When use .withConverter(), the httpClient from SupabaseClient() is not used;

To Reproduce

```dart
final postgrest = SupabaseClient('', '', httpClient: CustomHttpClient());
final User user = await postgrest
    .from('users')
    .select()
    .withConverter<User>((data) => User.fromJson(data));
```

Expected behavior
Should use CustomHttpClient()

### PR DESCRIPTION
**Pull request by Vinzent03**
fix https://github.com/supabase/supabase-flutter/issues/271

### CODE REVIEW

File: test/basic_test.dart [hunk 1]
```diff
@@ -399,6 +399,27 @@ void main() {
...
+            fail('Table was able to be selected, even tho it does not exist');
+          },
+          onError: (error) {
+            expect(error, isA<PostgrestException>());
+            expect(error.code, '420');
+          },
+        );
+      } on PostgrestException catch (error) {
+        expect(error.code, '420');
+      }
```
Comment (dshukertjr) [inline test/basic_test.dart]: Would we be able to do something like this instead?
```suggestion
        await postgrestCustomHttpClient
            .from('users')
            .select()
            .withConverter((data) => data);
            fail('Table was able to be selected, even tho it does not exist');
      } catch (error) {
        expect(error, isA<PostgrestException>());
        expect(error.code, '420');
      }
```

**Review by dshukertjr — COMMENTED** [review]
Great catch! I just had one thing that I would like to hear your opinion on, but everything else looks great!

**Comment by Vinzent03** [comment]
Great call, I simplified other cases as well.

**Review by dshukertjr — APPROVED** [review]
Looks great to me! Thanks for the amazing fix!

**Pull request merged**